agent_app_sample_code/utils/get_recursive_character_text_splitter.py

```diff
@@ -0,0 +1,214 @@
+# MAGIC - `chunk_size_tokens`: An optional size for each chunk in tokens. Defaults to `None`, which uses the model's entire context window.
+# MAGIC - `chunk_overlap_tokens`: Tokens that should overlap between chunks. Defaults to `0`.
+# MAGIC
+# MAGIC **Returns:** A callable that takes a document (`str`) and produces a list of chunks (`list[str]`).
+
+# COMMAND ----------
+
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from transformers import AutoTokenizer
+import tiktoken
+from typing import Callable, Tuple, Optional
+import os
+import re
+from databricks.sdk import WorkspaceClient
+
+# Constants
+HF_CACHE_DIR = "/tmp/hf_cache/"
+
+# Embedding Models Configuration
+EMBEDDING_MODELS = {
+    "gte-large-en-v1.5": {
+        "tokenizer": lambda: AutoTokenizer.from_pretrained(
+            "Alibaba-NLP/gte-large-en-v1.5", cache_dir=HF_CACHE_DIR
+        ),
+        "context_window": 8192,
+        "type": "SENTENCE_TRANSFORMER",
+    },
+    "bge-large-en-v1.5": {
+        "tokenizer": lambda: AutoTokenizer.from_pretrained(
+            "BAAI/bge-large-en-v1.5", cache_dir=HF_CACHE_DIR
+        ),
+        "context_window": 512,
+        "type": "SENTENCE_TRANSFORMER",
+    },
+    "bge_large_en_v1_5": {
+        "tokenizer": lambda: AutoTokenizer.from_pretrained(
+            "BAAI/bge-large-en-v1.5", cache_dir=HF_CACHE_DIR
+        ),
+        "context_window": 512,
+        "type": "SENTENCE_TRANSFORMER",
+    },
+    "text-embedding-ada-002": {
+        "context_window": 8192,
+        "tokenizer": lambda: tiktoken.encoding_for_model("text-embedding-ada-002"),
+        "type": "OPENAI",
+    },
+    "text-embedding-3-small": {
+        "context_window": 8192,
+        "tokenizer": lambda: tiktoken.encoding_for_model("text-embedding-3-small"),
+        "type": "OPENAI",
+    },
+    "text-embedding-3-large": {
+        "context_window": 8192,
+        "tokenizer": lambda: tiktoken.encoding_for_model("text-embedding-3-large"),
+        "type": "OPENAI",
+    },
+}
+
+
+def get_workspace_client() -> WorkspaceClient:
+    """Returns a WorkspaceClient instance."""
+    return WorkspaceClient()
+
+
+def get_embedding_model_config(endpoint_type: str) -> Optional[dict]:
+    """
+    Retrieve embedding model configuration by endpoint type.
+    """
+    return EMBEDDING_MODELS.get(endpoint_type)
+
+
+def extract_endpoint_type(llm_endpoint) -> Optional[str]:
+    """
+    Extract the endpoint type from the given llm_endpoint object.
+    """
+    try:
+        return llm_endpoint.config.served_entities[0].external_model.name
+    except AttributeError:
+        try:
+            return llm_endpoint.config.served_entities[0].foundation_model.name
+        except AttributeError:
+            try:
+                endpoint_name = llm_endpoint.config.served_entities[0].name
+                return (
```

Comment:
What is this case for? I don't fully get it

Reply:
lol good question. my best guess is that this would be for a custom embedding model.  im going to remove it since this is seems unlikely